File: project/core/module/rest-service/src/main/java/vdi/service/rest/config/SupportedArchiveType.java
package vdi.service.rest.config;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public enum SupportedArchiveType {
  ZIP(".zip"),
  TAR_GZ(".tgz", ".tar.gz");

  private final String[] fileExtensions;

  SupportedArchiveType(String... fileExtensions) {
    this.fileExtensions = fileExtensions;
  }

  public String[] getFileExtensions() {
    return fileExtensions;
  }

  public boolean matches(String filename) {
    for (String ext : fileExtensions) {
      if (filename.endsWith(ext))
        return true;
    }

    return false;
  }

  public static List<String> getAllSupportedExtensions() {
    var out = new ArrayList<String>(3);

    for (var entry : SupportedArchiveType.values())
      out.addAll(Arrays.asList(entry.fileExtensions));

    return out;
  }
}

File: project/core/module/rest-service/src/main/kotlin/vdi/service/rest/config/SupportedArchiveType.kt
package vdi.service.rest.config

@JvmInline
value class SupportedArchiveType private constructor(private val extensions: Array<String>) {
  fun matches(filename: String) = extensions.any(filename::endsWith)

  companion object {
    val ZIP
      get() = SupportedArchiveType(arrayOf(".zip"))

    val TAR_GZ
      get() = SupportedArchiveType(arrayOf(".tgz", ".tar.gz"))

    inline val entries
      get() = arrayOf(ZIP, TAR_GZ)

    fun getAllSupportedExtensions() = arrayOf(*ZIP.extensions, *TAR_GZ.extensions)
  }
}